File: examples/app-yaml-config/sdk/mjs/19-refresh.mjs
#!/usr/bin/env node
// F05 / story 04 — refreshConfig() against a temp endpoint.dev.yaml.
// Use parse-modify-write so the new endpoint lands inside the `endpoints:` map
// (in-place append wouldn't merge correctly because the file ends with intent_mapping:).
import { EndpointConfigSDK, loadConfigFromFile } from "@ployglot/app-yaml-fetch-config";
import { mkdtempSync, copyFileSync, readFileSync, writeFileSync } from "node:fs";
import { tmpdir } from "node:os";
import { fileURLToPath } from "node:url";
import { dirname, resolve, join } from "node:path";
import yaml from "js-yaml";

const HERE = dirname(fileURLToPath(import.meta.url));
const SRC = resolve(HERE, "../../../server/config/endpoint.dev.yaml");

const tmp = mkdtempSync(join(tmpdir(), "epcfg-"));
const tmpFile = join(tmp, "endpoint.dev.yaml");
copyFileSync(SRC, tmpFile);

loadConfigFromFile(tmpFile);
const sdk = new EndpointConfigSDK({ filePath: tmpFile });

const before = sdk.listKeys().sort();

const doc = yaml.load(readFileSync(tmpFile, "utf8"));
doc.endpoints.llm003 = {
  name: "Tertiary LLM",
  tags: ["llm", "tertiary"],
  baseUrl: "http://localhost:53000",
  method: "POST",
  headers: { "Content-Type": "application/json" },
  timeout: 30000,
  bodyType: "json",
};
writeFileSync(tmpFile, yaml.dump(doc));

sdk.refreshConfig();
const after = sdk.listKeys().sort();
const newKeys = after.filter((k) => !before.includes(k));

console.log(JSON.stringify({
  before_count: before.length,
  after_count: after.length,
  new_keys: newKeys,
  invariant_llm003_appeared: newKeys.includes("llm003"),
}, null, 2));

File: examples/app-yaml-config/sdk/py/19-refresh.py
#!/usr/bin/env python3
"""F05 / story 04 — refresh_config() against a temp endpoint.dev.yaml."""
from __future__ import annotations

import json
import shutil
import tempfile
from pathlib import Path

import yaml  # type: ignore[import-untyped]

from app_yaml_fetch_config import EndpointConfigSDK, load_config_from_file

HERE = Path(__file__).resolve().parent
SRC = (HERE / ".." / ".." / ".." / "server" / "config" / "endpoint.dev.yaml").resolve()

tmp_dir = Path(tempfile.mkdtemp(prefix="epcfg-"))
tmp_file = tmp_dir / "endpoint.dev.yaml"
shutil.copyfile(SRC, tmp_file)

load_config_from_file(str(tmp_file))
sdk = EndpointConfigSDK(file_path=str(tmp_file))

before = sorted(sdk.list_keys())

# Parse-modify-write so the new endpoint lands inside `endpoints:` (file ends with intent_mapping:).
doc = yaml.safe_load(tmp_file.read_text())
doc["endpoints"]["llm003"] = {
    "name": "Tertiary LLM",
    "tags": ["llm", "tertiary"],
    "baseUrl": "http://localhost:53000",
    "method": "POST",
    "headers": {"Content-Type": "application/json"},
    "timeout": 30000,
    "bodyType": "json",
}
tmp_file.write_text(yaml.safe_dump(doc))

sdk.refresh_config()
after = sorted(sdk.list_keys())
new_keys = [k for k in after if k not in before]

print(json.dumps({
    "before_count": len(before),
    "after_count": len(after),
    "new_keys": new_keys,
    "invariant_llm003_appeared": "llm003" in new_keys,
}, indent=2))
